docs/mindsdb_sql/agents/knowledge-bases.mdx

```diff
@@ -3,238 +3,436 @@ title: Knowledge Base
+A knowledge base is an advanced system designed to go beyond traditional data storage. Its components include embedding and reranking models and a vector store.
 
 <p align="center">
   <img src="https://docs.google.com/drawings/d/e/2PACX-1vTVmUqhoFXLF3ncL0NWwgdPjI7Hj19f-5xU8ED31ntYvlsTM3poM9zZKwcrEwzvxJrOhl2raIFKWlsp/pub?w=1342&h=681" />
 </p>
 
-## Syntax
+It intelligently organizes information in semantically meaningful ways, allowing for context-aware retrieval based on meaning rather than just keywords. With the ability to perform semantic reasoning across multiple pieces of information, it delivers deeper insights and more accurate responses.
 
-Here is a general syntax for creating a knowledge base:
+## Create a Knowledge Base
 
-```sql
+Here is the syntax for creating a knowledge base:
+
+<Note>
+MindsDB stores objects, such as models or knowledge bases, inside projects. The default project is called `mindsdb`. You can [create other projects](/mindsdb_sql/sql/create/project) to store your objects.
+
+To refer to an object stored in a defined project, specify both project and object names separated by a dot: `project_name.object_name`. For example, `my_project.my_kb`. If no project is defined with the object name, the object is stored in the default `mindsdb` project.
+</Note>
+
+<CodeGroup>
+
+```sql OpenAI
 CREATE KNOWLEDGE_BASE my_kb
 USING
-   model = embedding_model, -- optional
-   storage = vector_database.storage_table, -- optional
-   metadata_columns = ['date', 'creator', ...], -- optional
-   content_columns = ['review', 'content', ...], -- optional
-   id_column = 'index'; -- optional
+     embedding_model = {
+         "provider": "OpenAI",
+         "model_name" : "text-embedding-ada-002",
+         "api_key": "sk-..."  -- optional, default from env variable
+     },
+     reranking_model = {
+         "provider": "OpenAI",
+         "model_name": 'gpt-4o',
+         "api_key": "sk-..."  -- optional, default from env variable
+     },
+     storage = my_vector_store, -- optional, default ChromaDB
+     metadata_columns = ['date', 'creator', ...], -- optional
+     content_columns = ['review', 'content', ...], -- optional, default content
+     id_column = 'id'; -- optional
 ```
 
-Where all the parameters are optional:
+```sql OpenAI via Azure
+CREATE KNOWLEDGE_BASE my_kb
+USING
+     embedding_model = {
+         "provider": "Azure_OpenAI",
+         "model_name" : "text-embedding-ada-002",
+         "api_key": "sk-...",  -- optional, default from env variable
+         "azure_endpoint" : "https://ai-2249ai742481026689.openai.azure.com/",
+         "openai_api_version": "2024-12-01-preview"
+     },
+     reranking_model = {
+         "provider": "OpenAI",
+         "model_name": 'gpt-4o',
+         "api_key": "sk-..."  -- optional, default from env variable
+     },
+     storage = my_vector_store, -- optional, default ChromaDB
```

Comment:
@martyna-mindsdb This should be a table within a vector store, yes?

Reply:
Good catch. Fixed it.